pkg/k8s/clientFactory.go

```diff
@@ -23,6 +23,9 @@ type ClientFactory interface {
 	// CoreV1 returns the core/v1 Kubernetes client
 	CoreV1() corev1.CoreV1Interface
 
+	// KubernetesClientset returns Kubernetes clientset
+	KubernetesInterface() kubernetes.Interface
```

Comment:
I would use the name `KubernetesClientSet()` instead, as package `k8s.io/client-go/kubernetes` provides the Kubernetes client set, and `Interface` is just a _local_ name from this package.